packages/ui/src/utils/get-filtered-properties.ts

```diff
@@ -0,0 +1,33 @@
+import { KaotoSchemaDefinition } from '../models';
+import { isDefined } from './is-defined';
+
+/**
+ * Extracts the schema recursively containing only the filtered properties.
+ */
+export function getFilteredProperties(
+  properties: KaotoSchemaDefinition['schema']['properties'],
+  filter: string,
+  omitFields?: string[],
+): KaotoSchemaDefinition['schema']['properties'] {
+  if (!isDefined(properties)) return {};
+
+  const filteredFormSchema = Object.entries(properties).reduce(
+    (acc, [property, definition]) => {
+      if (!omitFields?.includes(property)) {
+        if (definition['type'] === 'object' && 'properties' in definition) {
+          const subFilteredSchema = getFilteredProperties(definition['properties'], filter);
```

Comment:
If there's a recursive schema, the browser might hang as there's no stop statement, maybe we could stop at the 3rd level for now, perhaps passing an index value and returning when reaching the 3rd level.

For instance, the `steps` property is a recursive one, but we're ignoring it in the form, I don't know if there could be another one.

Reply:
We agreed to kept it in its current state for the moment!